packages/wrangler/src/deployment-bundle/bundle.ts

```diff
@@ -311,6 +311,7 @@ export async function bundleWorker(
 		conditions: BUILD_CONDITIONS,
 		...(process.env.NODE_ENV && {
 			define: {
+				"navigator.userAgent": `"Cloudflare-Workers"`,
```

Comment:
Drive-by review so forgive me if I’m lacking any context, but isn’t this something set by the runtime? And more specifically not set with an old compat date or `no_global_navigator` flag enabled?

Let's say I'm importing a library that has code like this:

```javascript
function runCallback(callbackForWorkers, callbackForOthers){
	try {
		if (navigator.userAgent === 'Cloudflare-Workers') {
			return callbackForWorkers();
		}
		return callbackForOthers();
	} catch {
		return callbackForOthers();
	}
}
```

and then:

```javascript
export default {
	fetch(request) {
		runCallback(null, () => {
			return new Response(null);
		});
	}
}
```

This is a contrived example of course, but previously, if I was using an old compat date, this code would hit callbackForOthers, and then return a response. With this change to wrangler's build, without any code changes of my own, the next time I publish, the behaviour of my Worker could completely change.


If there's something I'm missing here, please let me know, but this seems like it could cause a lot of possible incompatibilites and breakages in userland code, which is something compat dates and flags are designed to prevent.